### PR TITLE
feat: add SSL support to RocketMQ connector

### PR DESCRIPTION
**Pull request by kjellwinblad**
Fixes https://emqx.atlassian.net/browse/EMQX-12289

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [] Added property-based tests for code which performs user input validation
- [] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [] For internal contributor: there is a jira ticket to track this change
- [] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

### CODE REVIEW

File: scripts/ct/run.sh [hunk 1]
```diff
@@ -202,6 +202,9 @@ for dep in ${CT_DEPS}; do
         rocketmq)
             FILES+=( '.ci/docker-compose-file/docker-compose-rocketmq.yaml' )
             ;;
+        rocketmq_ssl)
```
Comment (thalesmg) [inline scripts/ct/run.sh:205]: Nit: we could include this new docker compose file in the existing `rocketmq` entry.

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl [hunk 2]
```diff
@@ -113,13 +114,21 @@ on_start(
...
-
+    ClientCfg0 = #{acl_info => ACLInfo, namespace => Namespace},
+    SSLOpts = emqx_tls_lib:to_client_opts(SSLOptsMap),
+    ClientCfg =
+        case SSLOpts of
+            [] ->
+                ClientCfg0;
+            SSLOpts ->
+                ClientCfg0#{ssl_opts => SSLOpts}
+        end,
```
Comment (thalesmg) [inline apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl:125]: Micro nit:

```suggestion
    ClientCfg = emqx_utils_maps:put_if(ClientCfg0, ssl_opts, SSLOpts, SSLOpts =/= []),
```

File: apps/emqx_bridge_rocketmq/test/emqx_bridge_rocketmq_SUITE.erl [hunk 3]
```diff
@@ -180,6 +188,26 @@ create_bridge(Config) ->
...
+    BridgeType = ?GET_CONFIG(rocketmq_bridge_type, Config),
+    Name = ?GET_CONFIG(rocketmq_name, Config),
+    RocketMQConf = ?GET_CONFIG(rocketmq_config_ssl, Config),
+    emqx_bridge:create(BridgeType, Name, RocketMQConf).
+
+create_bridge_ssl_bad_ssl_opts(Config) ->
+    BridgeType = ?GET_CONFIG(rocketmq_bridge_type, Config),
+    Name = ?GET_CONFIG(rocketmq_name, Config),
+    RocketMQConf0 = ?GET_CONFIG(rocketmq_config_ssl, Config),
+    RocketMQConf1 = maps:put(
```
Comment (thalesmg) [inline apps/emqx_bridge_rocketmq/test/emqx_bridge_rocketmq_SUITE.erl:201]: Nit: maybe add a comment explaining what's wrong with this config?

**Review by thalesmg — APPROVED** [review]

File: scripts/ct/run.sh (hunk 1 above)
Comment (kjellwinblad) [reply scripts/ct/run.sh]: Fixed: [0cbbae9](https://github.com/emqx/emqx/pull/13274/commits/0cbbae9655fd28a1dd07bd553f274744e9b363de)

File: apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl (hunk 2 above)
Comment (kjellwinblad) [reply apps/emqx_bridge_rocketmq/src/emqx_bridge_rocketmq_connector.erl]: Fixed: [0cbbae9](https://github.com/emqx/emqx/pull/13274/commits/0cbbae9655fd28a1dd07bd553f274744e9b363de)

File: apps/emqx_bridge_rocketmq/test/emqx_bridge_rocketmq_SUITE.erl (hunk 3 above)
Comment (kjellwinblad) [reply apps/emqx_bridge_rocketmq/test/emqx_bridge_rocketmq_SUITE.erl:203]: Fixed: [0cbbae9](https://github.com/emqx/emqx/pull/13274/commits/0cbbae9655fd28a1dd07bd553f274744e9b363de)

**Review by thalesmg — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2646)

**Pull request merged**